### PR TITLE
OKD 4.4: install CRI-O from F31 modules

### PR DESCRIPTION
**Pull request by vrutkovs**
Enable CRI-O 1.16 module in cosa container and fetch RPMs via 
yumdownloader from DNF repos instead of RHEL8.

~~This also ensures e2e-aws test is being run when machine-os-content promotion happens~~

### CODE REVIEW

**Comment by vrutkovs** [comment]
```
Dec 16 13:04:34 ip-10-0-13-128 systemd[1]: Starting Container Runtime Interface for OCI (CRI-O)...
Dec 16 13:04:35 ip-10-0-13-128 crio[38721]: time="2019-12-16 13:04:35.018815676Z" level=fatal msg="Invalid character(s) found in prerelease \"module_f31\""
Dec 16 13:04:35 ip-10-0-13-128 systemd[1]: crio.service: Main process exited, code=exited, status=1/FAILURE
Dec 16 13:04:35 ip-10-0-13-128 systemd[1]: crio.service: Failed with result 'exit-code'.
```

**Comment by vrutkovs** [comment]
Reaches test phase: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/6415/rehearse-6415-promote-release-openshift-okd-machine-os-content-e2e-aws-4.4/5

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 1]
```diff
@@ -492,10 +494,14 @@ periodics:
               RUN set -xeuo pipefail && \\
                   mkdir -p /srv/repo /tmp/working && \\
                   curl -L "${tar_url}" | tar xf - -C /srv/repo/ && \\
+                  cp /etc/yum.repos.d/fedora-updates-testing-modular.repo /tmp && \\
```
Comment (vrutkovs) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: TODO: Host repo files similar to `base-4-x.svc` instead of re-using repo files from cosa container

**Comment by vrutkovs** [comment]
```
error: unable to check whether to include image registry.svc.ci.openshift.org/ci-op-2ff3e3dd/machine-os-content@sha256:7872b1b33e0b28fe4e6610945bd5c0122598e0bf30cfe20b409016aad7b99651: tag "machine-os-content" references a component version "machine-os" which is not in the allowed list
```
/retest

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 2]
```diff
@@ -457,10 +459,10 @@ periodics:
         latest="$( oc get istag -n fcos ${TAG}:${BRANCH} --template '{{ index .image.dockerImageMetadata.Config.Labels "version" }} {{ index .image.dockerImageMetadata.Config.Labels "id-artifacts" }} {{ index .image.dockerImageMetadata.Config.Labels "id-machine-config-operator-rpms" }}' )"
         candidate="${build_id} ${origin_artifacts_id} ${origin_machine_config_operator_rpms_id}"
         echo "Comparing existing '$latest' against latest '$candidate'"
-        if [[ "${latest}" == "${candidate}" ]]; then
-          echo "No build necessary, content is up to date"
-          exit 0
-        fi
+        # if [[ "${latest}" == "${candidate}" ]]; then
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: Uncomment this.

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 3]
```diff
@@ -512,13 +514,24 @@ periodics:
...
+                  if ! grep -q cri-o /tmp/packages; then yumdownloader -y --disablerepo=* --enablerepo=updates-testing-modular --destdir=/tmp/rpms cri-o cri-tools; fi && \\
                   curl http://base-${BRANCH//\./-}-rhel8.ocp.svc > /etc/yum.repos.d/rhel8.repo && \\
-                  if ! grep -q cri-o /tmp/packages; then yumdownloader -y --disablerepo=* --enablerepo=rhel-8* --destdir=/tmp/rpms cri-o cri-tools; fi && \\
-                  if ! grep -q glusterfs-fuse /tmp/packages; then yumdownloader -y --disablerepo=* --enablerepo=rhel-8* --destdir=/tmp/rpms attr glusterfs glusterfs-client-xlators glusterfs-fuse glusterfs-libs psmisc; fi && \\
+                  if ! grep -q glusterfs-fuse /tmp/packages; then yumdownloader -y --disablerepo=* --enablerepo=rhel-8* --destdir=/tmp/rpms glusterfs glusterfs-client-xlators glusterfs-fuse glusterfs-libs psmisc; fi && \\
                   cd /tmp/working && \\
                   for i in \$(find /tmp/rpms/ -name *.rpm); do echo "Extracting \$i ..."; rpm2cpio \$i | cpio -div; done && \\
+                  cd usr/sbin && \
+                  ln -s xtables-nft-multi iptables && \\
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: This looks weird.  Why is it necessary?

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 4]
```diff
@@ -538,10 +551,30 @@ periodics:
         oc -n ${namespace} create is machine-os-content >/dev/null
         build ${dir}/build.yaml machine-os-content
 
+        # Craft a release image
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: Split this change into its own PR (along with the rehearse stuff).

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml (hunk 3 above)
Comment (vrutkovs) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: By default Fedora 31 points to legacy iptables. This would switch alternatives to nft version (as in RHCOS). This however doesn't seem to help, so I'll keep it and use this PR to produce debug versions of OKD payload for now

**Comment by vrutkovs** [comment]
Removed rehearse and AWS test part to unblock this:

* Install CRI-O 1.16 from F31 modules
* Use oc from 4.4 nightly
* attr is already included in FCOS

File: ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml [hunk 5]
```diff
@@ -512,11 +512,15 @@ periodics:
               RUN set -xeuo pipefail && \\
                   mkdir -p /srv/repo /tmp/working && \\
                   curl -L "${tar_url}" | tar xf - -C /srv/repo/ && \\
+                  cp /etc/yum.repos.d/fedora-updates-testing-modular.repo /tmp && \\
                   rm -rf /etc/yum.repos.d/* && \\
+                  cp /tmp/fedora-updates-testing-modular.repo /etc/yum.repos.d/ && \\
+                  sed -i 's/enabled=0/enabled=1/g' /etc/yum.repos.d/fedora-updates-testing-modular.repo && \\
                   rpm-ostree db list --repo /srv/repo "${ostree_ref}" > /tmp/packages && \\
+                  dnf module enable -y cri-o:1.16 && \\
```
Comment (smarterclayton) [inline ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: it needs to be crio 1.17 now

Comment (vrutkovs) [reply ci-operator/jobs/openshift/release/openshift-release-release-4.4-periodics.yaml]: There hasn't been cri-o 1.17 release yet - and no 1.17 module yet. I'll be pinging @lsm5 to get module created and update this task
